### PR TITLE
intl: increase coverage for NumberFormat.format

### PR DESCRIPTION
**Pull request by ryzokuken**
Increase test coverage for Intl.NumberFormat.prototype.format by adding
a check for testing if it calls ToNumber (7.1.3) on its argument.

/cc @rwaldron @gsathya @littledan @anba

### CODE REVIEW

File: test/intl402/NumberFormat/prototype/format/value-tonumber.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  [false, 0],
+  ['42', 42],
+  ['foo', NaN]
+];
+
+const nf = new Intl.NumberFormat();
+
+toNumberResults.forEach(pair => {
+  const value = pair[0];
+  const result = pair[1];
```
Comment (gsathya) [inline test/intl402/NumberFormat/prototype/format/value-tonumber.js]: Use destructuring to make this cleaner

`([value, result]) => { ... }`

File: test/intl402/NumberFormat/prototype/format/value-tonumber.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+esid: sec-number-format-functions
+description: >
+  Tests that Intl.NumberFormat.prototype.format converts its argument
+  (called value) to a number using ToNumber (7.1.3).
+info: |
+  11.1.4Number Format Functions
+
+  4. Let x be ? ToNumber(value).
+features: [Symbol]
+---*/
```
Comment (gsathya) [inline test/intl402/NumberFormat/prototype/format/value-tonumber.js]: Ideally you'd want a test that patches @@toPrimitive to make sure that the ToNumber is actually being called

**Comment by ryzokuken** [comment]
Done. @gsathya PTAL.

File: test/intl402/NumberFormat/prototype/format/value-tonumber.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+const nf = new Intl.NumberFormat();
+
+toNumberResults.forEach(pair => {
+  const [value, result] = pair;
+  assert.sameValue(nf.format(value), nf.format(result));
+});
+
+let count = 0;
+const dummy = {};
+dummy[Symbol.toPrimitive] = hint => (hint === 'number' ? ++count : NaN);
```
Comment (gsathya) [inline test/intl402/NumberFormat/prototype/format/value-tonumber.js]: @@toPrimitive takes two args: input, hint

Comment (ryzokuken) [reply test/intl402/NumberFormat/prototype/format/value-tonumber.js]: @gsathya I was referring to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol/toPrimitive, which only took one argument. Isn't the input stored at `this` anyway?

**Comment by ryzokuken** [comment]
ping @rwaldron

**Pull request merged**